study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EventBuilder.java

```diff
@@ -79,9 +89,45 @@ private void insertEvents(Handle handle) {
         }
     }
 
+    private void insertLabeledEvents(Handle handle, String[] eventLabelArray) {
+        if (!cfg.hasPath("events")) {
+            return;
+        }
+        Set<String> eventLabelsToLoad = new HashSet<>(Arrays.asList(eventLabelArray));
+        List<Config> allLabeledCfgs = cfg.getConfigList("events").stream()
+                .filter(eventCfg -> eventCfg.hasPath("label") && eventCfg.getString("label") != null)
+                .collect(toList());
+       List<String> allCfgLabels = allLabeledCfgs.stream().map(cfg -> cfg.getString("label")).collect(toList());
+       Set<String> notFoundEventLabels = eventLabelsToLoad.stream().filter(eventName -> !allCfgLabels.contains(eventName)).collect(toSet());
+       if (!notFoundEventLabels.isEmpty()) {
+           throw new DDPException("Could not find events " + String.join(", ", notFoundEventLabels));
+       }
+       if (allCfgLabels.size() > eventLabelsToLoad.size()) {
+           throw new DDPException("Found duplicate names in event configuration entries");
+       }
+       List<Config> eventCfgsToLoad = allLabeledCfgs.stream()
+                   .filter(cfg -> allCfgLabels.contains(cfg.getString("label")))
+                   .collect(toList());
+
+       List<String> existingCfgLabelsInDb = eventCfgsToLoad.stream()
+                .map(eventCfg -> handle.attach(EventDao.class)
+                        .getAllEventConfigurationsByStudyIdAndLabel(studyDto.getId(), eventCfg.getString("label")))
```

Comment:
Okay, I see how this query is being used. In this case, how about we make the query take in a set of labels to query and return a list/stream of those that are found?

Reply:
Thought of doing query, then decided this was easier to do. For seldom executed code iterating over a handful of items ought to be fine.